air/src/utils.rs

```diff
@@ -1,8 +1,172 @@
 use alloc::vec::Vec;
-use core::ops::Range;
+use core::ops::{Add, AddAssign, Index, IndexMut, Mul, Range, Rem, Shr, Sub};
 
 use super::FieldElement;
-use vm_core::utils::range as create_range;
+use vm_core::{utils::range as create_range, Felt};
+
+// NEWTYPE WRAPPERS
+// ================================================================================================
+
+/// A newtype wrapper around a usize value representing a step in the execution trace.
+#[derive(Debug, Copy, Clone)]
+pub struct Step(u32);
+
+// FROM STEP
+
+impl From<Step> for u32 {
+    fn from(step: Step) -> u32 {
+        step.0
+    }
+}
+
+impl From<Step> for i32 {
+    fn from(step: Step) -> i32 {
+        step.0 as i32
+    }
+}
+
+impl From<Step> for usize {
+    fn from(step: Step) -> usize {
+        step.0 as usize
+    }
+}
+
+impl From<Step> for Felt {
+    fn from(step: Step) -> Felt {
+        Felt::from(step.0)
+    }
+}
+
+// INTO STEP
+
+impl From<usize> for Step {
+    fn from(value: usize) -> Step {
+        Step(value as u32)
```

Comment:
I think we should use a `value.try_into().expect()` to catch when the `usize` value is too large. The assumption is that we would ensure elsewhere that traces are never longer than `2^32 - 1`, and that we should never hit that panic. But in the case that we'd do, I would prefer to panic over silently wrapping around

Reply:
Do we want to panic on overflow for addition and multiplication?

Reply:
yes I think that's fine, at least for this PR, as this is the current behavior (since we're using primitive types)